stdlib/src/utils/static_tuple.mojo

```diff
@@ -110,7 +110,8 @@ fn _static_tuple_construction_checks[size: Int]():
     Parameters:
       size: The number of elements.
     """
-    constrained[size > 0, "number of elements in `StaticTuple` must be > 0"]()
```

Comment:
Is it intended to disabling this compile-time check?

Reply:
Indeed. We need for this PR to work to create a InlineArray of size 0, because we can't conditionally declare fields in a struct.

Reply:
Maybe the check can be kept as >= 0?